### ISSUE TITLE
Clicking "Cancel" for a connection with "Ask every time" password will connect when it should cancel/fail

### ISSUE THREAD

**Issue by skplunkerin** [TablePlus/TablePlus#2867]
### Please answer the questions below, it helps us to keep track of the issue. Please do not ignore it or your issue will be closed.

1. Which driver are you using and version of it (Ex: PostgreSQL 10.0):
   - PostgreSQL 14.5

2. Which TablePlus build number are you using (the number on the welcome screen, Ex: build 81):
   - Version: 5.1.0
   - Build: 468
   - OS: macOS Monterey (Version 12.6)

3. The steps to reproduce this issue:
   1. edit a PostgreSQL connection:
      - delete the `Password` field value
      - change the `Password` dropdown to `Ask every time`
      - click `Save`
   2. try canceling a connection password prompt (works first time):
      - double click the connection to open the connection
      - you'll see the password prompt
      - click `Cancel`
      - this works, you can close the window
   3. connect to the connection:
      - double click the connection to open the connection
      - enter your password in the prompt
      - this works, close the window
   4. **[HERE'S THE BUG]** try canceling a connection password prompt _(breaks on 2nd+ attempts)_:
      - double click the connection to open the connection
      - you'll see the password prompt
      - click `Cancel`
      - the connection doesn't cancel, instead it connects to the database without the password
      - exit the window and edit the connection:
         - notice how the `Password` field value is set when it should not be set, I believe a previous successful connection is setting this field and ignoring the `Ask every time` logic?

---

Basically, a connection with the password set to `Ask every time` will store the password on a successful connection, and 2nd+ connection requests will allow you to connect to the connection without needing the password; that defeats the purpose of `Ask every time` _(although, it does ask every time... you can just skip/ignore it and still connect which is not the expected behavour)_.
